src/main/java/bio/terra/service/auth/iam/IamResourceType.java

```diff
@@ -39,10 +47,12 @@ static IamResourceType fromValue(String text) {
   }
 
   public static IamResourceType fromEnum(IamResourceTypeEnum apiEnum) {
-    if (apiEnum == IamResourceTypeEnum.SPEND_PROFILE) {
-      return IamResourceType.SPEND_PROFILE;
+    for (IamResourceType b : IamResourceType.values()) {
+      if (b.getIamResourceTypeEnum().equals(apiEnum)) {
+        return b;
+      }
     }
-    return fromString(apiEnum.toString());
+    return null;
```

Comment:
why do we want to throw `null` and not an Error?

Reply:
I don't know! That was an existing pattern, but tbh I'm just gunna have it throw